src/include/migraphx/op/scatternd_op.hpp

```diff
@@ -36,36 +37,71 @@ namespace op {
-                           " is too large for tensor of rank " + std::to_string(r));
-        if(not(std::equal(ind_lens.begin(), ind_lens.begin() + q - 1, upd_lens.begin()) and
-               std::equal(data_lens.begin() + k, data_lens.end(), upd_lens.begin() + q - 1)))
-            MIGRAPHX_THROW("ScatterND: incorrect update shape. update.lens != indices.lens[0:q-1] "
-                           "++ data.lens[k:r-1]");
-        auto s = inputs.front();
-        if(s.broadcasted())
+        check_shapes{inputs, *this, true}.has(3);
+
+        if(migraphx::none_of(inputs, [](auto v) { return v.dynamic(); }))
         {
-            return {s.type(), s.lens()};
+            auto r         = inputs.front().lens().size();
+            auto q         = inputs.at(1).lens().size();
+            auto k         = inputs.at(1).lens().back();
+            auto ind_lens  = inputs.at(1).lens();
+            auto upd_lens  = inputs.back().lens();
+            auto data_lens = inputs.front().lens();
+
+            if(q + r - ind_lens.back() - 1 != upd_lens.size())
+            {
+                char msg[100];
+                sprintf(msg,
+                        "ScatterND:  ranks of inputs don't match. %lu + %lu - %lu - 1 != %lu\n",
+                        q,
+                        r,
+                        ind_lens.back(),
+                        upd_lens.size());
+                MIGRAPHX_THROW(msg);
+            }
+
+            if(k > r)
+                MIGRAPHX_THROW("ScatterND: index of size " + std::to_string(k) +
+                               " is too large for tensor of rank " + std::to_string(r));
+            if(not(std::equal(ind_lens.begin(), ind_lens.begin() + q - 1, upd_lens.begin()) and
+                   std::equal(data_lens.begin() + k, data_lens.end(), upd_lens.begin() + q - 1)))
+                MIGRAPHX_THROW(
+                    "ScatterND: incorrect update shape. update.lens != indices.lens[0:q-1] "
+                    "++ data.lens[k:r-1]");
+            auto s = inputs.front();
+            if(s.broadcasted())
+            {
+                return {s.type(), s.lens()};
+            }
+            else
+            {
+                return s.with_lens(s.lens());
+            }
         }
         else
         {
+            // One or more inputs are dynamic; shape validation can't be done yet.  Putting off
+            // validation until 3 static shapes are received means that if any inputs are dynamic
+            // shapes, the validation will happen at evaluation time.
+            auto s = inputs.front();
+            if(s.dynamic())
+                return s;
+            if(s.broadcasted())
+                return {s.type(), s.lens()};
             return s.with_lens(s.lens());
         }
     }
 
-    argument compute(const shape& output_shape, std::vector<argument> args) const
+    /** The operation's computation.  Populate the output with the data, and scatter the updates
+     * into it.*/
```

Comment:
This comment is not very descriptive to me. Maybe remove it. If scatternd works differently than the ONNX op you can describe the differences at the top of this file. Otherwise can leave blank.